projects/nx-verdaccio/src/plugin/targets/environment.targets.unit-test.ts

```diff
@@ -0,0 +1,463 @@
+import {
+  EXECUTOR_ENVIRONMENT_BOOTSTRAP,
+  VERDACCIO_REGISTRY_JSON,
+} from '../../executors/env-bootstrap/constants';
+import { EXECUTOR_ENVIRONMENT_TEARDOWN } from '../../executors/env-teardown/constants';
+import { EXECUTOR_ENVIRONMENT_SETUP } from '../../executors/env-setup/constants';
+
+import * as nodePathModule from 'node:path';
+import * as uniquePortModule from '../../executors/env-bootstrap/unique-port';
+
+const JOIN_RESULT = 'mocked-join';
+const PROJECT_NAME = 'unit-test-project';
+const ENVIRONMENTS_DIRECTORY = '/environments';
+const TARGET_NAMES = ['e2e'];
+const TAGS = ['env:production'];
+const TARGETS: { [targetName: string]: TargetConfiguration } = {
+  e2e: { executor: 'nx:test', options: {} },
+  build: { executor: 'nx:build', options: {} },
+};
+const PROJECT_CONFIG: ProjectConfiguration = {
+  root: 'mock-root',
+  name: PROJECT_NAME,
+  targets: TARGETS,
+  tags: [...TAGS],
+};
+const OPTIONS: NormalizedCreateNodeOptions['environments'] = {
+  environmentsDir: ENVIRONMENTS_DIRECTORY,
+  targetNames: TARGET_NAMES,
+  filterByTags: TAGS,
+};
+
+describe('isEnvProject', (): void => {
+  it('should return false if targets are missing', (): void => {
+    const config = { ...PROJECT_CONFIG, targets: null };
+    const result = isEnvProject(config, OPTIONS);
+    expect(result).toBe(false);
+  });
+
+  it('should return false if targetNames are missing', (): void => {
+    const options = { ...OPTIONS, targetNames: null };
+    const result = isEnvProject(PROJECT_CONFIG, options);
+    expect(result).toBe(false);
+  });
+
+  it('should return false if targetNames and targets are missing', (): void => {
+    const config = { ...PROJECT_CONFIG, targets: null };
+    const options = { ...OPTIONS, targetNames: null };
+    const result = isEnvProject(config, options);
+    expect(result).toBe(false);
+  });
+
+  it('should return false if targetNames don’t match environmentTargetNames', (): void => {
+    const options = { ...OPTIONS, targetNames: ['mockTarget'] };
+    const result = isEnvProject(PROJECT_CONFIG, options);
+    expect(result).toBe(false);
+  });
+
+  it('should return true if targetNames match and tags are not present', (): void => {
+    const config = { ...PROJECT_CONFIG, tags: null };
+    const result = isEnvProject(config, OPTIONS);
+    expect(result).toBe(true);
+  });
+
+  it('should return true if targetNames match and filterByTags are not present', (): void => {
+    const options = {
+      ...OPTIONS,
+      filterByTags: null,
+    };
+    const result = isEnvProject(PROJECT_CONFIG, options);
+    expect(result).toBe(true);
+  });
+
+  it('should return true if targetNames match and tags match filterByTags', (): void => {
+    const result = isEnvProject(PROJECT_CONFIG, OPTIONS);
+    expect(result).toBe(true);
+  });
+
+  it('should return false if targetNames match but tags don’t match filterByTags', (): void => {
```

Comment:
In the tests return values and `true` and `false` are mixed. We could start with the happy path and list all results of `true` first and then the ones with `false`